var/spack/repos/builtin/packages/py-awkward0/package.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyAwkward0(PythonPackage):
```

Comment:
The py-awkward0 package was released February 8, 2021, which seems pretty recent. I am pretty sure that py-awkward0 is correct.

Reply:
@glennpj No, the developers are very clear on this one:

>  Fundamental changes were also required to streamline remote reading (HTTP and XRootD), so Uproot 4 was distributed as a separate project in parallel with Uproot 3 (like Awkward 1 and 0). For the latter half of 2020, adventurous users and downstream developers could install uproot as a separate project.
https://raw.githubusercontent.com/scikit-hep/uproot4/main/docs-img/diagrams/uproot-awkward-timeline.png

> On December 1, 2020, however, Awkward 0 and Uproot 3 were deprecated, moved to PyPI packages awkward0 and uproot3, while Awkward 1 and Uproot 4 became unqualified as awkward and uproot.

https://uproot.readthedocs.io/en/latest/uproot3-to-4.html

If you check the commit history of awkward0.x, it  is clear that there are only fixes to help people who need it for legacy code, not new developments. I'd say that if possible, it would be a good idea to avoid adding the old version and see if the downstream package can migrate to the new uproot/awkward. If not, please add a comment that this is a legacy package and that `awkward`/`uproot` should be used for new projects.

Reply:
Here is what I get when using py-uproot4 with gatetools:
```console
$ gt_gate_info 
uproot3 is mandatory to merge root file. Please, do:
pip install uproot3
uproot3 is mandatory to merge root file. Please, do:
pip install uproot3
Gate path:          /opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-broadwell/gcc-8.4.0/gate-9.0-npreog6wphf5eaenfhkuqyp5ezmnqyut/bin/Gate
Version:            Gate version is 9.0 ; Geant4 version is 10.6.3
...
```
So, gatetools needs py-uproot3, and py-uproot3 and py-uproot3-methods need py-awkward0. I am not involved with the OpenGate project. I am just trying to provide spack packages for researchers who use gate and gatetools, so I am not in a position to say whether the gatetools project is or is not looking into updating that dependency. Since the error message above specifically calls out uproot3 it seems that they are aware of the situation. It seems to me that this is a legitimate use of the legacy packages provided by PyPI and Spack should provide them as needed.

Reply:
Sure, that's what uproot3 is for. I had a quick look and gatetools uses the write functionality in uproot3 that is still not completely ported to uproot, so it is also not trivial to replace it for the developers at the moment.